Feat/remove contribution

oxlime opened the pull request:
This is what i have so far, my idea of removing by basically zero'ing out the contribution. I'm having issues with removing while also looping through the contributions.

AnthonyBuisset commented on contracts/onlydust/deathnote/core/contributions/test_contributions.cairo:
```diff
@@ -53,6 +53,33 @@ func test_new_contribution_can_be_added{
+    %{ stop_prank() %}
+
+    let (count, contribs) = contributions.all_contributions()
+
```
you can also check that if you try to read the deleted contribution, the call is reverted with "Contributions: Contribution does not exist"

AnthonyBuisset reviewed (changes requested):
Nice ! Thanks for the contribution.

Could you please modify the e2e test as well ? Check `test_contributions.e2e.cairo` file.
It should show that a change is missing in the Interface as well.

I do not think there is a better way of doing it as it is not possible to delete storage in starknet AFAIK. Zero-ing all fields is the way to do.
Good work overall !

oxlime commented:
Deleted contributions are still indexed but all values set to 0. So I reworked the fetch_contribution_loop to loop over contributions with contribution_id = 0.

oxlime commented:
closing this because i opened a new pr after the refactor
